app/controllers/collections_controller.rb

```diff
@@ -27,42 +27,39 @@ def index
     if params[:work_id] && (@work = Work.find_by(id: params[:work_id]))
       @collections = @work.approved_collections.by_title.includes(:parent, :moderators, :children, :collection_preference, owners: [:user]).paginate(page: params[:page])
     elsif params[:collection_id] && (@collection = Collection.find_by(name: params[:collection_id]))
-      @collections = @collection.children.by_title.includes(:parent, :moderators, :children, :collection_preference, owners: [:user]).paginate(page: params[:page])
+      @collections = CollectionSearchForm.new({ parent_id: @collection.id }).search_results.to_a.paginate(page: params[:page])
     elsif params[:user_id] && (@user = User.find_by(login: params[:user_id]))
-      @collections = @user.maintained_collections.by_title.includes(:parent, :moderators, :children, :collection_preference, owners: [:user]).paginate(page: params[:page])
+      @collections = CollectionSearchForm.new(moderator_ids: [@user.id]).search_results.to_a.paginate(page: params[:page])
       @page_subtitle = ts("%{username} - Collections", username: @user.login)
     else
       if params[:user_id]
```

Comment:
Let's remove all these `flash.now[:error]` checks. Instead can you change the 3 `find_by`s into `find_by!`s to raise 404 errors?